packages/client/lib/CredentialRequestClientBuilder.ts

```diff
@@ -40,24 +23,31 @@ export class CredentialRequestClientBuilder {
     version?: OpenId4VCIVersion;
     credentialIdentifier?: string;
     credentialTypes?: string | string[];
-  }): CredentialRequestClientBuilder {
-    const issuer = credentialIssuer;
-    const builder = new CredentialRequestClientBuilder();
-    builder.withVersion(version ?? OpenId4VCIVersion.VER_1_0_11);
-    builder.withCredentialEndpoint(metadata?.credential_endpoint ?? (issuer.endsWith('/') ? `${issuer}credential` : `${issuer}/credential`));
-    if (metadata?.deferred_credential_endpoint) {
-      builder.withDeferredCredentialEndpoint(metadata.deferred_credential_endpoint);
-    }
-    if (credentialIdentifier) {
-      builder.withCredentialIdentifier(credentialIdentifier);
-    }
-    if (credentialTypes) {
-      builder.withCredentialType(credentialTypes);
+  }): CredentialRequestClientBuilderV1_0_13 | CredentialRequestClientBuilderV1_0_11 {
+    const specVersion = version ? version : OpenId4VCIVersion.VER_1_0_13;
+    if (specVersion >= OpenId4VCIVersion.VER_1_0_13) {
+      return CredentialRequestClientBuilderV1_0_13.fromCredentialIssuer({
+        credentialIssuer,
+        metadata,
+        version,
+        credentialIdentifier,
+        credentialTypes,
+      });
+    } else {
+      if (!credentialTypes || credentialTypes.length === 0) {
+        throw new Error('CredentialTypes must be provided for v1_0_11');
```

Comment:
Why doesn't the v11 builder take care of this?

Reply:
this is because in the `redentialRequestClientBuilderV1_0_11.fromCredentialIssuer` the `credentialTypes` param is required and in the draft13 it is optional. so I think it's better for it to be optional in the general `fromCredentialIssuer` function and we only check if it's passing a value or not.